strings/longest_substr_without_repeating_chars.cpp

```diff
@@ -0,0 +1,105 @@
+/**
+ * @file
+ * @brief Greedy algorithm to find length of longest substring without repeating
+ * characters
+ *
+ * @details
+ * We are given a string. We need to compute the length of the longest
+ * substring, such that no character in the substring occurs more than once.
+ *
+ * For a string of length n, the time complexity of this algorithm is O(n)
+ * (amortized)
+ *
+ * @author [Nishant Mahesh](https://github.com/nishant-mahesh)
+ */
+
+#include <algorithm>  // To use the inbuilt 'max' function
+#include <cassert>
+#include <iostream>
+#include <map>
+#include <string>
+
+using namespace std;
+
+/**
+ * This function takes a string as input, and returns the length of the longest
+ * substring that contains no repeated characters
```

Comment:
Missing parameter documentation.

Reply:
What exactly does this mean? Do you mean I must specify that this function takes a string variable named **input** as input?

Reply:
> What exactly does this mean? Do you mean I must specify that this function takes a string variable named **input** as input?

You must add documentation for the parameter: what does it do, how does it work, etc..
You can take this code as a reference. I hope that helps. Let us know if you need any more help! 😃

```c++
/**
 * Utility function to print matrix
 * @tparam n number of matrix size
 * @param board matrix where numbers are saved
 */
template <size_t n>
void printSolution(const std::array<std::array<int, n>, n> &board) {
    std::cout << "\n";
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            std::cout << "" << board[i][j] << " ";
        }
        std::cout << "\n";
    }
}
```